pkg/reconciler/ingress/ingress.go

```diff
@@ -77,6 +80,20 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) r
 		}
 		if ready {
 			external, internal := config.ServiceHostnames()
+
+			if ingressconfig.FromContextOrDefaults(ctx).Kourier.TrafficIsolation == config.IsolationIngressPort {
+				ns, err := r.namespaceLister.Get(ing.Namespace)
+				if err != nil {
+					return fmt.Errorf("failed to get namespace: %w", err)
+				}
+
+				if ns.Annotations != nil {
+					if port, ok := ns.Annotations[config.ListenerPortAnnotationKey]; ok {
+						internal = config.ListenerServiceHostnames(port)
+					}
+				}
+			}
```

Comment:
This could now be extracted into a function since it's shared with `lister.go` 90-110.

(For a future PR)

Reply:
I'll do this in a future PR. (added to the list of TODOs)